### PR TITLE
use new etcd client and also enable godep

### PR DESCRIPTION
**Pull request by AdoHe**
I just enable godep, but very not sure about this.

### CODE REVIEW

**Comment by miekg** [comment]
I don't like godep, maybe gb? Or use the new go vender stuff?

**Comment by miekg** [comment]
can't look at the diff due to godep. Can you pull that out?

**Comment by AdoHe** [comment]
just remove godep. I will fix the tests problem.

File: main.go [hunk 1]
```diff
@@ -206,50 +187,52 @@ func validateHostPort(hostPort string) error {
...
-		u, e := url.Parse(m.Value)
-		if e != nil {
-			continue
-		}
-		// etcd=bla&raft=bliep
-		// TODO(miek): surely there is a better way to do this
-		ms := strings.Split(u.String(), "&")
-		if len(ms) == 0 {
-			continue
+func newHttpTransport(tlsCertFile, tlsKeyFile, tlsCACertFile string) etcd.CancelableTransport {
```
Comment (miekg) [inline main.go]: newHTTPSTransport ?

File: server/server_test.go [hunk 2]
```diff
@@ -302,7 +308,7 @@ func TestDNSTtlRRset(t *testing.T) {
 
 	ttl := uint32(60)
 	for _, serv := range services {
-		addService(t, s, serv.Key, uint64(ttl), serv)
+		addService(t, s, serv.Key, 60, serv)
 		defer delService(t, s, serv.Key)
 		ttl += 60
```
Comment (miekg) [inline server/server_test.go]: is ttl still used here?

File: main.go [hunk 3]
```diff
@@ -206,50 +187,52 @@ func validateHostPort(hostPort string) error {
...
+		Transport: newHttpTransport(tlsCert, tlsKey, tlsCACert),
+	}
+	cli, err := etcd.New(etcdCfg)
+	if err != nil {
+		return nil, err
 	}
-	return etcd.NewClient(machines)
+
+	// Periodically sync etcd
+	go cli.AutoSync(ctx, 15*time.Second)
```
Comment (miekg) [inline main.go]: what does this do? Thought the syncing was needed anymore in new(er) clients. Good be mistaken though.

**Pull request merged**